src/mots/directory.py

```diff
@@ -0,0 +1,254 @@
+        if frmt not in supported_formats:
+            raise ValueError(f"{frmt} not one of {supported_formats}.")
+
+        return getattr(self, f"_export_{frmt}")()
+
+
+class QueryResult:
+    """Helper class to simplify query result interpretation."""
+
+    paths: list = None
+    path_map: dict = None
+    rejected_paths: list = None
+    modules: list = None
+    owners: list = None
+    peers: list = None
+
+    data_keys = {
+        "paths",
+        "rejected_paths",
+        "modules",
+        "owners",
+        "peers",
+    }
+
+    def __init__(self, result, rejected):
+        data = {k: list() for k in self.data_keys}
+        self.path_map = result
+
+        for path in self.path_map:
+            data["paths"].append(path)
+            data["modules"] += self.path_map[path]
+
+        for module in data["modules"]:
+            # TODO: this conversion should happen elsewhere.
+            data["owners"] += [Person(**o) for o in module.owners]
+            data["peers"] += [Person(**p) for p in module.peers]
+
+        data["rejected_paths"] = rejected
+
+        # Remove duplicate entries in all data attributes.
+        for key in data:
+            setattr(self, key, list(set(data[key])))
+
+    def __add__(self, query_result):
+        """Merge the data from both QueryResult objects."""
+        path_map = self.path_map.copy()
+        path_map.update(query_result.path_map)
+        rejected_paths = set.intersection(
+            set(self.rejected_paths),
+            set(query_result.rejected_paths),
+        )
+        return QueryResult(path_map, rejected_paths)
+
+    def __radd__(self, query_result):
+        """Call self.__add__ since the order of addition does not matter."""
+        return self.__add__(query_result)
+
+
+@dataclass
+class Person:
+    """A class representing a person."""
+
+    bmo_id: int = None
+    name: str = ""
+    info: str = ""
+    nick: str = ""
+    bmo_data: InitVar[dict] = None
+
+    def __post_init__(self, bmo_data):
+        """Refresh BMO data from BMO API."""
+        if bmo_data:
+            self.nick = bmo_data.get("nick", "")
+            self.bmo_id = bmo_data.get("id") or self.bmo_id
+            real_name = bmo_data.get("real_name", "")
+
+            parsed_real_name = parse_real_name(real_name)
+            self.name = parsed_real_name["name"]
+            self.info = parsed_real_name["info"]
+
+    def __hash__(self):
+        """Return a unique identifier for this person."""
+        return int(self.bmo_id)
+
+
+class People:
+    """A people directory searchable by name, email, or BMO ID."""
+
+    by_bmo_id: dict = None
+    people: list = None
+    serialized: list = None
+
+    def __init__(self, people, query_bmo: bool = True):
+        logger.debug(f"Initializing people directory with {len(people)} people...")
+        if query_bmo:
+            bmo_client = BMOClient()
+            bmo_data = bmo_client.get_users_by_ids([p["bmo_id"] for p in people])
```

Comment:
For better test ergonomics, I'd recommend moving "work that touches the outside" (such as HTTP requests) to a separate function.
For example, turn `People` into a `@dataclass`, but give it a `def from_bmo()` function that production code can use.